README.md

```diff
@@ -259,10 +259,22 @@ It's often useful to plug some custom steps into the testing environment. Custom
 tests short and self-explanatory and at the same time help teams to add generic assertions that are meaningful in their 
 environment.
 
-To add custom steps in YAKS, you can fork + clone the [yaks extension](https://github.com/nicolaferraro/yaks-extension) repository, that provides
-an example of how to do that.
+To add custom steps in YAKS, you can look at the example provided in the [examples/extension](/examples/extension) directory.
+The example consists of a feature file ([examples/extension/extension.feature](/examples/extension/extension.feature)) using a custom step from a local project 
+([examples/extension/steps](/examples/extension/steps)).
 
-You can add your own steps to that project and follow the instructions in order to install them in the YAKS environment.
+To run the example:
+
+```
+yaks test extension.feature -u steps/
+```
+
+The `-u` flag stands for "upload". The steps project is built before running the test and the artifacts are uploaded to a
+[Snap](https://github.com/container-tools/snap) Minio server, in order for the test to retrieve them
+when needed.
+This happens transparently to the user.
+
+The local library can also be uploaded to the Snap Minio server prior to running the test, using the `yaks upload` command.  
```

Comment:
When using the `upload` command the user needs to give the path to the `artifact.jar` right? Would be good to have a code example for that

Reply:
No, the jar currently is not supported directly. It works with a maven project. The implementation is basic at the moment, but can be improved in snap: https://github.com/container-tools/snap/blob/5ed5ff1a76f523039c729695781348b87870847d/pkg/language/java/java.go#L31

It makes a release in a local path and then uploads everything to the minio server.

Reply:
Support for the jar can be added later

Reply:
Ok now I see. This is part of the snap code base. I was searching for the project build in this repo. That explains everything. Thanks!